src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -762,6 +763,50 @@ namespace FanCoilUnits {
                     ShowContinueError(state, "..Fan coil unit air outlet node name = " + state.dataLoopNodes->NodeID(FanCoil(FanCoilNum).AirOutNode));
                     ErrorsFound = true;
                 }
+                bool ZoneNodeNotFound = true;
+                for (CtrlZone = 1; CtrlZone <= state.dataGlobal->NumOfZones; ++CtrlZone) {
+                    if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZone).IsControlled) continue;
+                    for (NodeNum = 1; NodeNum <= state.dataZoneEquip->ZoneEquipConfig(CtrlZone).NumExhaustNodes; ++NodeNum) {
+                        if (FanCoil(FanCoilNum).AirInNode == state.dataZoneEquip->ZoneEquipConfig(CtrlZone).ExhaustNode(NodeNum)) {
+                            ZoneNodeNotFound = false;
+                            break;
+                        }
+                    }
+                    if (!ZoneNodeNotFound) break;
+                }
+                if (ZoneNodeNotFound) {
+                    // Varify zone number and return air node
+                    for (CtrlZone = 1; CtrlZone <= state.dataGlobal->NumOfZones; ++CtrlZone) {
+                        for (int Num = 1; Num <= state.dataZoneEquip->ZoneEquipList(CtrlZone).NumOfEquipTypes; ++Num) {
+                            if (UtilityRoutines::SameString(FanCoil(FanCoilNum).Name, state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipName(Num)) &&
+                                UtilityRoutines::SameString(FanCoil(FanCoilNum).UnitType,
+                                                            state.dataZoneEquip->ZoneEquipList(CtrlZone).EquipType(Num))) {
+                                FanCoil(FanCoilNum).ControlZoneNum = CtrlZone;
+                                break;
+                            }
+                        }
+                        if (FanCoil(FanCoilNum).ControlZoneNum > 0) break;
+                    }
+                    bool InletNodeFound = false;
+                    if (FanCoil(FanCoilNum).ControlZoneNum > 0) {
+                        for (NodeNum = 1; NodeNum <= state.dataZoneEquip->ZoneEquipConfig(FanCoil(FanCoilNum).ControlZoneNum).NumReturnNodes;
+                             ++NodeNum) {
+                            InletNodeFound = ZonePlenum::ValidateInducedNode(
+                                state, FanCoil(FanCoilNum).AirInNode, state.dataZoneEquip->ZoneEquipConfig(CtrlZone).ReturnNode(NodeNum));
+                            if (InletNodeFound) break;
+                        }
+                    }
```

Comment:
This block is testing to see of one if a zone return node is connected to the same plenum as the induced air node? Is this necessary?
This would be more efficient if the full list `state.dataZoneEquip->ZoneEquipConfig(CtrlZone).ReturnNode` is passed to the function. Then the function can first search for a plenum that matches the induced air node, then run through the return node list checking for a match there.
Also, if this fails because the induced node happens to be connected to a different plenum that does not return from the same zone, then the error message below will be confusing.

Reply:
@mjwitte If a full list of return node is passed as an argument, return node do loop will occur inside the function. Therefore, do loop will be needed in either outside function or inside function. I can make change. By the way, how can I place a full list of return nodes as a single argument?